### ISSUE TITLE
Bug: unaccessible menu buttons due to notifications

### ISSUE THREAD

**Issue by geromegrignon** [open-sauced/insights#1098]
### Describe the bug

By updating the profile information and clicking on 'Update profile' button, a notification appears in the left right corner of the screen.
But due to its padding (4rem), the header buttons are not accessible anymore as the notification uses `z-index: 100`.

An example with the colored padding:
<img width="423" alt="image" src="https://user-images.githubusercontent.com/32737308/232040553-80a5f817-70c7-4deb-bc42-6b85d652f6dc.png">

## Solution

It would be better to use `top: 4rem` to set the position of the notification.
The `z-index`value should be updated too: it's currently **100** but **10** for the dropdown menu of the header. By still having the notification visible, the menu is hidden behind.

### Steps to reproduce

1. go to https://insights.opensauced.pizza/user/settings
2. update an information
3. click on update profile button
4. try to click on the icon avatar in the header

### Affected services

insights.opensauced.pizza

### Platforms

Desktop

### Browsers

Chrome

### Environment

Production, Development

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by a0m0rajab** [comment]
I just checked it fastly, as for the padding maybe it can be removed and the top property be set to 4rem. 
As for the menu behind it, changing its z-index to 10 is a great idea,  

Here is the related code: 

https://github.com/open-sauced/insights/blob/beta/components/atoms/Toast/toast.tsx#L16 

For open sauced team, I would like to fix this, what would you suggest as a solution?

**Comment by a0m0rajab** [comment]
.take

**Comment by a0m0rajab** [comment]
i will work on this issue during this week

**Comment by a0m0rajab** [comment]
I think this is intentional, and you have to close the notification before doing anything else! 

if it's a bug I have two thoughts: 
- should the user be able to open the profile? Not sure about the best design for this one. 
- The notification could be on the bottom as well. 

I would like to have some input from the open sauced team. 

Here is how it looks on mobile: 
![image](https://user-images.githubusercontent.com/18273833/233185386-d7ca6148-af49-4118-a4a0-3789ab4f33e5.png)

**Comment by bdougie** [comment]
Curious to hear your thoughts here @OgDev-01

**Comment by OgDev-01** [comment]
Hey @a0m0rajab Thanks for picking this up. We should move the toast a little down below the nav on mobile screens 👍

 Around here
<img width="357" alt="toast-demo" src="https://user-images.githubusercontent.com/62995161/233272469-7b2d7df9-36d6-4737-8c06-310e70a3b97c.png">

Users should be able to open their profile even then toast it visible.

**Comment by a0m0rajab** [comment]
@OgDev-01 Thank you for the information. I just started working on it but got stuck. Here is my update: 

Removing the padding and adding the top position to the toast solves the issue. Yet, I got stuck on another issue: when the user is in the middle of the page. Having the previous design is better unless we have a way in tailwind to check if the page is on top or middle (not sure how)

Here are the screenshots: 
Good behavior: 
![image](https://user-images.githubusercontent.com/18273833/233501662-3c570c2d-99c8-46dd-a2d8-4f2562c63815.png)
The bad behavior: 
![image](https://user-images.githubusercontent.com/18273833/233501954-e9c941fd-e2ea-4113-9f6d-d84e5dc0ccc7.png)

Another solution we can have is to keep the same position but change the Z-index to let the user use the profile menu. 


The code: 
at https://github.com/open-sauced/insights/blob/beta/components/atoms/Toast/toast.tsx#L16
Removed: 
```
p-6 lg:p-16 sm:top-auto
```
Added: 
```
top-24 md:top-14
```

Plus, the Ipad Mini's behavior is a bit odd: 
![image](https://user-images.githubusercontent.com/18273833/233502485-74d6bc0c-0095-4edd-bc63-fa4427f5be0e.png)

Might need to open an issue for it.

**Comment by a0m0rajab** [comment]
Just did a Google search to find other examples: 

Here is the HP guide: https://design-system.hpe.design/templates/toast-notifications 
They do hide the links behind it.

**Comment by OgDev-01** [comment]
>Removing the padding and adding the top position to the toast solves the issue. Yet, I got stuck on another issue: when the user is in the middle of the page. 

Toasts are obstructive in nature and I don't think we should be concerned about any element that is blocked except the nav on this. 


This work's, but you will have to include `px-6` to move it out of the edge 👍 
![image](https://user-images.githubusercontent.com/18273833/233501662-3c570c2d-99c8-46dd-a2d8-4f2562c63815.png)

**Comment by a0m0rajab** [comment]
Thank you for the information!

Just to make sure doing this will have a space like the second picture I shared (the timezone one) is that acceptable?

**Comment by a0m0rajab** [comment]
I am assuming the like in the last comment is a yes. 
Did not get a notification about it, though. 

I did change the code. Here are the results
![image](https://user-images.githubusercontent.com/18273833/234838338-7fd6b449-3ba7-479c-a5e7-071ff1988198.png)

With the nav menu:
![image](https://user-images.githubusercontent.com/18273833/234838526-68129ed5-6666-4346-a3a9-ebf2b592b906.png)

Large screens: 
![image](https://user-images.githubusercontent.com/18273833/234838590-d0c74fbf-6499-4b6c-9fc6-843e9cf95e1b.png)


The toast can still be in the upper place in the mobile and hide the nav behind it. 

Side note: px is not the best for a position. I went with top to not hide the elements behind it.
